bouncer/shared/utils.ts

```diff
@@ -294,84 +294,107 @@ export async function observeSwapEvents(
   { sourceAsset, destAsset, depositAddress, channelId }: SwapParams,
   api: ApiPromise,
   tag?: string,
-  swapType?: SwapType,
   finalized = false,
 ): Promise<BroadcastId | undefined> {
   let eventFound = false;
   const subscribeMethod = finalized
     ? api.rpc.chain.subscribeFinalizedHeads
     : api.rpc.chain.subscribeNewHeads;
 
+  const swapRequestedEvent = 'SwapRequested';
   const swapScheduledEvent = 'SwapScheduled';
   const swapExecutedEvent = 'SwapExecuted';
   const swapEgressScheduled = 'SwapEgressScheduled';
   const batchBroadcastRequested = 'BatchBroadcastRequested';
-  let expectedMethod = swapScheduledEvent;
+  let expectedMethod = swapRequestedEvent;
 
-  let swapId = 0;
+  let swapId: number | undefined;
+  let swapRequestId: number | undefined;
   let egressId: EgressId;
   let broadcastId;
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const unsubscribe: any = await subscribeMethod(async (header) => {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const events: any[] = await api.query.system.events.at(header.hash);
-    events.forEach((record) => {
-      const { event } = record;
-      if (!eventFound && event.method.includes(expectedMethod)) {
+
+    // Some events don't appear in the order one might expect (e.g. SwapRequested
+    // comes after SwapScheduled), so as a workaround, we add an other loop to be
+    // able to re-scan events that we already visited:
+    let keepSearching = true;
+
+    while (keepSearching) {
+      keepSearching = false;
+
+      for (const record of events) {
+        const { event } = record;
+        if (eventFound || !event.method.includes(expectedMethod)) {
+          // eslint-disable-next-line no-continue
+          continue;
+        }
+
         const expectedEvent = {
           data: event.toHuman().data,
         };
 
+        const data = expectedEvent.data;
+
         switch (expectedMethod) {
```

Comment:
Didn't write this code, but I'm guessing it is because the name is the event is exposed as "method" through substrate's API. I will change to `expectedEvent` to be consistent with the other vars.